code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,782 @@
+            "EnvironmentCoordinates": info_ET["EnvironmentCoordinates"],
+            "ScreenAOIDefinition": info_ET["ScreenAOIDefinition"],
+            "SamplingFrequency": sampling_frequency,
+            "StartTime": timestamp_first_trigger,
+            "RecordedEye": recorded_eye,
+            "EyeTrackingMethod": eye_tracking_method,
+            "PupilFitMethod": pupil_fit_method,
+            "GazeMappingSettings": {
+                "CRThreshold": CR_threshold,
+                "PThreshold": pupil_threshold,
+            },
+            "CalibrationCount": calibration_count,
+            "Columns": self.samples.columns.tolist(),
+            "CalibrationType": calibration_type,
+            "CalibrationPosition": calibration_position,
+            "AverageCalibrationError": average_calibration_error,
+            "MaximalCalibrationError": max_calibration_error,
+            "EDFHeader": header,
+        }
+
+        final_info = {
+            key: value for key, value in final_info.items() if value is not None
+        }
+
+        if not final_info:
+            print("No data to write to the JSON file. Skipping.")
+            return None
+
+        if self.task_name in ["rest", "bht", "qct"]:
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_dir-{self.pe}_eyetrack.json"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/func/",
+            )
+        elif self.task_name == "dwi":
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_acq-highres_dir-{self.pe}_eyetrack.json"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/dwi/",
+            )
+        else:
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_eyetrack.json"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/func/",
+            )
+
+        os.makedirs(output_file_dir, exist_ok=True)
+        output_json_path = os.path.join(output_file_dir, output_file_name)
+        with open(output_json_path, "w") as json_file:
+            json.dump(final_info, json_file, indent=4)
+
+        return output_json_path
+
+    def plot_pupil_size(
+        self,
+        eye: str = "right",
+        notebook: bool = True,
+        save: bool = False,
+        path_save: str = ".",
+        filename: Optional[str] = None,
+    ) -> Optional[str]:
+        if filename is None:
+            filename = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_pupil_ts.pdf"
+
+        if eye == "right":
+            self.samples.pa_right[self.samples.pa_right < 1] = np.nan
+            plt.plot(self.samples["time"], self.samples["pa_right"])
```

Comment:
Following my comment on the notebook, I think it would be more stable to send numpy arrays instead of pandas series to `plt.plot`.
```suggestion
            plt.plot(self.samples["time"].values, self.samples["pa_right"].values)
```